public/util/lex.ts

```diff
@@ -91,62 +107,203 @@ class DistilRelationState extends RelationState {
   const suggestions = variablesToLexSuggestions(filteredVariables);
+  // this generates the base templates used for the user typing into the lexbar
+  const baseSuggestion = variablesToLexSuggestions(
+    filteredAllVariables.map((v) => {
+      return { variable: v, count: 1, mode: INCLUDE_FILTER };
+    })
+  );
+
   const catVarLexSuggestions = perCategoricalVariableLexSuggestions(
-    filteredVariables
+    filteredAllVariables.map((v) => {
+      return v;
+    })
   );
+  const allSuggestions = [...suggestions, ...baseSuggestion];
   return Lex.from("field", ValueState, {
     name: "Choose a variable to search on",
     icon: '<i class="fa fa-filter" />',
-    suggestions: suggestions,
+    suggestions: allSuggestions,
   }).branch(
     Lex.from("relation", DistilRelationState, {
       ...TransitionFactory.valueMetaCompare({ type: TEXT_FILTER }),
     }).branch(Lex.from("value", TextEntryState)),
-    Lex.from("relation", DistilRelationState, {
-      ...TransitionFactory.valueMetaCompare({ type: CATEGORICAL_TYPE }),
-    }).branch(
-      Lex.from("value", ValueState, {
-        allowUnknown: false,
-        icon: "",
-        name: "Type for suggestions",
-        fetchSuggestions: (hint, lexDefintion) => {
-          return catVarLexSuggestions[lexDefintion.field.key]
-            ? catVarLexSuggestions[lexDefintion.field.key].filter((cat) => {
-                return cat.key.toLowerCase().indexOf(hint.toLowerCase()) > -1;
-              })
-            : [];
-        },
-      })
-    ),
-    Lex.from("relation", DistilRelationState, {
-      ...TransitionFactory.valueMetaCompare({ type: NUMERICAL_FILTER }),
-    }).branch(
-      Lex.from(LabelState, { label: "From" })
-        .to("min", NumericEntryState, { name: "Enter lower bound" })
-        .to(LabelState, { label: "To" })
-        .to("max", NumericEntryState, { name: "Enter upper bound" })
-    ),
-    ...distilDateTimeEntryBuilder(suggestions),
-    Lex.from("relation", DistilRelationState, {
-      ...TransitionFactory.valueMetaCompare({ type: GEOBOUNDS_FILTER }),
-    }).branch(
-      Lex.from(LabelState, { label: "From Latitude" })
-        .to("minX", NumericEntryState, { name: "Enter lower bound" })
-        .to(LabelState, { label: "To" })
-        .to("maxX", NumericEntryState, { name: "Enter upper bound" })
-        .to(LabelState, { label: "From Longitude" })
-        .to("minY", NumericEntryState, { name: "Enter lower bound" })
-        .to(LabelState, { label: "To" })
-        .to("maxY", NumericEntryState, { name: "Enter upper bound" })
-    )
+    ...distilCategoryEntryBuilder(allSuggestions, catVarLexSuggestions),
+    ...distilNumericalEntryBuilder(allSuggestions),
+    ...distilDateTimeEntryBuilder(allSuggestions),
+    ...distilGeoBoundsEntryBuilder(allSuggestions)
   );
 }
+export function distilCategoryEntryBuilder(
+  suggestions: ValueStateValue[],
+  catVarLexSuggestions: Dictionary<unknown[]>
+): StateTemplate[] {
+  const categoryEntries = [];
```

Comment:
As declared, this variable is effectively untyped, and will be treated as an `any[]` wherever it is used.  If the type is knowable, it should be declared as `const categroyEntries = [] as Foo[]`, which will allow the type checker to to deal with it appropriately.  Same applies to declarations of `{}`.  There are a few similar instances in this file that should be fixed if the type is known.